build-tools/packages/build-cli/src/library/markdown.ts

```diff
@@ -26,18 +27,63 @@ export function remarkHeadingLinks(): (tree: Node) => void {
 			if (node.children?.length > 0) {
 				const firstChild = node.children[0];
 				if (firstChild.type === "text") {
-					const headingValue = firstChild.value;
-					const slug = slugger.slug(headingValue);
+					const slug = slugger.slug(toString(node));
 					// We need to insert an Html node instead of a string, because raw
 					// strings will get markdown-escaped when rendered
 					const htmlNode: Html = {
 						type: "html",
 						value: `<a id="${slug}"></a>`,
 					};
-					// Insert the HTML node before the text node of the heading
+					// Insert the HTML node as the first child node of the heading
 					node.children.unshift(htmlNode);
 				}
 			}
 		});
 	};
 }
+
+/**
+ * A regular expression that extracts an admonition title from a string UNLESS the admonition title is the only thing on
+ * the line.
+ *
+ * Capture group 1 is the admonition type/title. Capture group 2 is any trailing whitespace.
+ *
+ * @remarks
+ *
+ * Description of the regular expression:
+ *
+ * This regular expression matches patterns in the form of `[!WORD]` where WORD can be CAUTION, IMPORTANT, NOTE, TIP, or
+ * WARNING. It ensures that the pattern is not followed by only whitespace characters until the end of the line.
+ * Additionally, it captures any whitespace characters that follow the matched pattern.
+ */
+const ADMONITION_REGEX = /(\[!(?:CAUTION|IMPORTANT|NOTE|TIP|WARNING)])(?!\s*$)(\s*)/gm;
```

Comment:
Two things about the regex:

- Do we need to capture the trailing whitespace? Doesn't seem to be used.
- Would be nice to add a few tests to make sure it matches only what we want.

Reply:
Good point on the whitespace. I tested in regexr, but I can look at unit tests too. The test story is just so annoying for build-tools atm. :)